flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompositeSerializer.java

```diff
@@ -0,0 +1,258 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.typeutils;
+
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.core.memory.DataInputView;
+import org.apache.flink.core.memory.DataOutputView;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nonnull;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Base class for composite serializers.
+ *
+ * <p>This class serializes a composite type using array of its field serializers.
+ * Fields are indexed the same way as their serializers.
+ *
+ * @param <T> type of custom serialized value
+ */
+public abstract class CompositeSerializer<T> extends TypeSerializer<T> {
+	private static final long serialVersionUID = 1L;
+
+	protected final TypeSerializer<Object>[] fieldSerializers;
+	final boolean isImmutableTargetType;
```

Comment:
I think in Java code style, a boolean field name should not be prefixed with `is...`, only the getter should be prefixed with `is...`